src/editor/Renderer.cpp

```diff
@@ -1557,7 +1657,104 @@ void Renderer::setupView() {
 	view.translate(-cameraOrigin.x, -cameraOrigin.z, cameraOrigin.y);
 }
 
+void Renderer::reloadBspModels()
+{
+	isModelsReloading = true;
+
+	if (!mapRenderers.size())
+	{
+		isModelsReloading = false;
+		return;
+	}
+
+	int modelcount = 0;
+
+	for (int i = 0; i < mapRenderers.size(); i++)
+	{
+		if (mapRenderers[i]->map->is_model)
+		{
+			modelcount++;
+		}
+	}
+
+	if (modelcount == mapRenderers.size())
+	{
+		isModelsReloading = false;
+		return;
+	}
+
+	std::vector<BspRenderer*> sorted_renders;
+
+	for (int i = 0; i < mapRenderers.size(); i++)
+	{
+		if (!mapRenderers[i]->map->is_model)
+		{
+			sorted_renders.push_back(mapRenderers[i]);
+		}
+		else
+		{
+			delete mapRenderers[i];
+		}
+	}
+
+	mapRenderers = sorted_renders;

```

Comment:
I'm having a hard time following this. It looks like all map renderers are replaced with a list of renderers that are for models only. Wouldn't that delete the primary map renderer?

Reply:
@wootguy I think this code remove models from main render, many time elapsed since I edit code, I don't know what's happening if I do remove it...